### PR TITLE
Variables deletion reloaded

### LINKED ISSUE #897: Deleting all variables doesn't work if one variable is derived from another (in GUI).
In the UVCDAT GUI, I had a few variables and wanted to delete them all.  Hence, I clicked the select_all, and hit delete.  However, it produced an error that some variables were derived from other variables (which was true because I had generated variables  that were averages of the original variables).   This raises two issues:  (1) I expected that the new variable would be independent of the ones it is derived from, which makes me worry that I have to worry about variable dependencies, and (2) why should it matter if I am deleting all of the variables?   Note: I am using version 2.0.0.

### PR DESCRIPTION
**Pull request by hgohil2805**
- Works for various levels of dependencies between variables. 
- removed the trash all method as it's not being used anywhere

Fixes UV-CDAT/uvcdat#897

### CODE REVIEW

**Comment by hgohil2805** [comment]
@remram44 Can you please make sure everything is OK?

@cameronsmith1 , @doutriaux1  can you please try this branch?

File: vistrails/gui/uvcdat/definedVariableWidget.py [hunk 1]
```diff
@@ -449,14 +451,33 @@ def clear(self):
...
+        while(varDeleted > 0):
+          varDeleted = 0 
+          for v in self.getSelectedDefinedVariables():
+              returnValue = self.deleteVariable(v.id,count)
+              if(returnValue):
+                varDeleted = varDeleted + 1
+        count = 2
         for v in self.getSelectedDefinedVariables():
-            self.deleteVariable(v.id)
+            self.deleteVariable(v.id,count)
```
Comment (remram44) [inline vistrails/gui/uvcdat/definedVariableWidget.py]: What is this second loop for?

Comment (hgohil2805) [inline vistrails/gui/uvcdat/definedVariableWidget.py]: Second loop is to generate the popup errors of dependency if there are any
in the final list..

Comment (remram44) [inline vistrails/gui/uvcdat/definedVariableWidget.py]: Can you replace "count" "1" and "2" with a readable parameter?

Comment (hgohil2805) [inline vistrails/gui/uvcdat/definedVariableWidget.py]: OK I will.

**Comment by remram44** [comment]
I had to rewrite the whole thing but here we go.

**Pull request merged**